src/new_particle_filter.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+from numpy.random import uniform
+from numpy.random import randn
+# For random samples from N(\mu, \sigma^2), use:
+# sigma * np.random.randn(...) + mu
+from numpy.linalg import norm
+from relative_dubins_dynamics import RelativeDubinsDynamics
+from filterpy.monte_carlo import systematic_resample
+import scipy.stats
+import matplotlib.pyplot as plt
+import pdb
+
+
+class ParticleFilter(object):
+
+    def __init__(self):
+        self.StateTransitionFcn = None
+        self.MeasurementLikelihoodFcn = None
+        self.particles = None
+        self.weights = None
+
+    def rot(self, theta):
+        return np.matrix(
+            [[np.cos(theta), -np.sin(theta)], [np.sin(theta), np.cos(theta)]])
+
+    # bounds is d x 2 where d is dimensions of a particle
+    def create_uniform_particles(self, N, bounds, circ_var):
+        dim = np.shape(bounds)[0]
+
+        particles = np.asmatrix(np.empty((N, dim)))
+
+        for i in np.arange(dim):
+            particles[:, i] = np.asmatrix(
+                uniform(bounds[i, 0], bounds[i, 1], size=N)).T
+            if circ_var[i] == 1:
+                particles[:, i] %= 2 * np.pi
+
+        self.particles = particles
+        self.weights = np.asmatrix(np.ones((N, 1)))
+        self.weights /= sum(self.weights)
+
+    def correct(self, *args):
+        np.multiply(self.weights, self.MeasurementLikelihoodFcn(
+            self.particles, *args))
+
+        self.weights += 1.e-100
+        self.weights /= sum(self.weights)
+
+        N = np.shape(self.weights)[0]
+        if self.neff(self.weights) < N/2:
+            indexes = systematic_resample(self.weights)
+            self.particles, self.weights = self.resample_from_index(
+                self.particles, self.weights, indexes)
+
+        return self.estimate(self.particles, self.weights)
+
+    # u should be a column matrix
```

Comment:
Comments should be under first line of function starting with three quotation marks

Reply:
You can also use an assert here.